csharp/ql/lib/semmle/code/cil/InstructionGroups.qll

```diff
@@ -73,8 +73,8 @@ class ComparisonOperation extends BinaryExpr, @cil_comparison_operation {
 class BinaryArithmeticExpr extends BinaryExpr, @cil_binary_arithmetic_operation {
   override Type getType() {
     exists(Type t0, Type t1 |
-      t0 = this.getOperand(0).getType().getUnderlyingType() and
```

Comment:
Maybe adding a comment onto `Expr::getType()` mentioning that there's an alternative `getOperandType` which should be used instead of `getOperand(X).getType()` would be beneficial. 

Should all instances be changed?

I found that `ex.getExpr().getType()` also calls `getOperand(X).getType` in https://github.com/github/codeql/blob/c9f1c9839065d5b63bd6084d64f209ed417cf6eb/csharp/ql/lib/semmle/code/cil/CallableReturns.qll#L28